lib/AST/Type.cpp

```diff
@@ -2491,6 +2501,15 @@ ReplaceOpaqueTypesWithUnderlyingTypes::operator()(CanType maybeOpaqueType,
   
   auto archetype = archetypeAndRoot->first;
   auto opaqueRoot = archetypeAndRoot->second;
+
+  // Don't replace opaque types from resilient modules.
```

Comment:
This bit of code is copy and pasted twice, please turn it into a method on ReplaceOpaqueTypesWithUnderlyingTypes. Also it should take the resilience expansion into account:

- If the opaque type is from an inlinable function, replace opaque type with the underlying type.
- If expansion is Maximal and we're in the same module, replace the opaque type with the underlying type.
- If expansion is Minimal and we didn't hit the first case, do nothing.